### PR TITLE
Feature #2558 tc_diag_docs

### LINKED ISSUE #2558: Revise TC-Diag documentation for version 11.1.0
## Describe the Task ##
#2168 added the initial version of the TC-Diag tool. While it does include some initial documentation, it is not thorough or completely accurate. This task is to review and revise that documentation prior to the 11.1.0 release.

### Time Estimate ###
4 hours.

### Sub-Issues ###
Consider breaking the task down into sub-issues.
None needed.

### Relevant Deadlines ###
Version 11.1.0 release.

### Funding Source ###
2770043

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
- [x] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by JohnHalleyGotway**
## Expected Differences ##

- [x] Do these changes introduce new tools, command line arguments, or configuration file options? **[No]**</br>
If **yes**, please describe:</br>

- [x] Do these changes modify the structure of existing or add new output data types (e.g. statistic line types or NetCDF variables)? **[No]**</br>
If **yes**, please describe:</br>

## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Built the docs locally to ensure that they render well with no warnings.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
  - Confirm that all automated tests runs with no differences flagged.
  - Please review the [feature_2558_tc_diag_docs](https://met.readthedocs.io/en/feature_2558_tc_diag_docs/Users_Guide/tc-diag.html) branch of the docs. Please review for typos and clarity. And please feel free to make any updates you'd like, particularly in the [Introduction](https://met.readthedocs.io/en/feature_2558_tc_diag_docs/Users_Guide/tc-diag.html#introduction) section.
  - I added several "notes" to indicate that TC-Diag is still under development as of MET version 11.1.0.
  - Note that this PR includes updates to the TC-Diag chapter and also updates the flowchart for version 11.1.0.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**
None needed.

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[Thursday 6/22/2023]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Development** issue with the original issue number.
- [x] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [x] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Comment by JohnHalleyGotway** [comment]
@jvigh thanks for making these updates to the docs in [this commit](https://github.com/dtcenter/MET/pull/2580/commits/c1539625919222ef1b3e84e0128831a8231a360b). I reviewed them via [RTD here](https://met--2580.org.readthedocs.build/en/2580/Users_Guide/tc-diag.html#practical-information).

I see that you:
1. Updated the intro section. Thanks for adding that background info. It reads well to me.
2. Added sub-sections to the config file section. We don't have this in the other chapters, but I do think this is a good change. That way we can provide direct links to specific parts of the config file. @jprestop please feel free to weigh in with any other feedback about this.

Note that I did just commit one edit to fix the number of underline characters used on line 104. You were short by one which threw and warning message and caused the documentation GHA workflow to fail. I expect the docs to render without warnings after that change.

**Comment by jvigh** [comment]
@johnhg Yeah, I was reading this and it said "subsections below", but then
there were no headings. The only clue to readers that there was a new
subsection were the horizontal rule lines. Then the reader would have to
just dig into the text below the config section to try to understand what
it was about. So I added headings. Modify them if they're too wordy or if
there's a better name for the config section.

**Comment by jprestop** [comment]
@jvigh and @JohnHalleyGotway, I also think that the addition of the subsection headers is a good change.  Not only does it now provide direct links to specific parts of the config file, but it also seems more readable as well.  Thanks @jvigh!

**Comment by jprestop** [comment]
Also, I added a comment to an existing [issue](https://github.com/dtcenter/MET/issues/1862) to consider making the formatting the same across all config options for consistency.

**Review by jvigh — COMMENTED** [review]
In this paragraph: 

tc_diag_STORMID_ATCFID_YYYYMMDDHH_cyl_grid_DOMAIN.nc where STORMID is the 2-letter basin name, 2-digit storm number, and 4-digit year, ATCFID is the track model name, YYYYMMDDHH is the track initialization time, and DOMAIN is the domain name.

The ATCF nomenclature would be to call ATCFID 'TECH', where TECH is described as the "acronym for each objective technique", which could simply be called ATCF TECH ID for these purposes. But if MET uses ATCFID elsewhere, I think it's fine to leave as is. If not, it might be good to change ATCFID to TECH. 

https://www.nrlmry.navy.mil/atcf_web/docs/database/new/abdeck.txt

**Review by jvigh — APPROVED** [review]
I have noted in a comment re: ATCFID vs. the ATCF's actual nomenclature of TECH. Apart from that, everything looks good. I will defer to @JohnHalleyGotway whether any changes are needed.

**Comment by JohnHalleyGotway** [comment]
Finalizing these TC-Diag documentation updates after...
1. Replacing "ATCFID" with "TECH".
2. Changing the destination branch from develop to main_v11.1.
